clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/services/AsgService.groovy

```diff
@@ -57,6 +58,29 @@ class AsgService {
     } ?: null
   }
 
+  List<AbstractServerGroupNameResolver.TakenSlot> getTakenSlots(String clusterName) {
+    new AwsResultsRetriever<AutoScalingGroup, DescribeAutoScalingGroupsRequest, DescribeAutoScalingGroupsResult>() {
+      @Override
+      protected DescribeAutoScalingGroupsResult makeRequest(DescribeAutoScalingGroupsRequest request) {
+        amazonAutoScaling.describeAutoScalingGroups(request)
+      }
+
+      @Override
+      protected List<AutoScalingGroup> accessResult(DescribeAutoScalingGroupsResult result) {
+        result.autoScalingGroups.findAll { AutoScalingGroup asg ->
+          def names = Names.parseName(asg.autoScalingGroupName)
+          return names.cluster == clusterName
+        }
+      }
+    }.retrieve(new DescribeAutoScalingGroupsRequest()).collect { AutoScalingGroup asg ->
```

Comment:
I'm going to merge this down as I'm working in a near identical area to avoid the need to make an empty DescribeAutoScalingGroupsRequest.

In one of our large regions, this takes upwards of 2 minutes+ to finish processing.

I've got changes that make the previous ASG lookup go against the cache ... PR will be coming shortly once I've incorporated these changes.